app/controllers/page_controller.rb

```diff
@@ -3,6 +3,15 @@ def show
     page_slug = params[:page_slug] ? params[:page_slug] : 'home'
     @page = Page.includes(:page_group).find_by(slug: page_slug.downcase, page_groups: {slug: params[:page_group_friendly_id].downcase})
     @page_components = @page.page_components
+    @page_components.each do |pc|
+      if pc.component_type == "PageMapComponent"
+        @map_component = PageMapComponent.find_by_id(pc.component_id)
+        @short_name = @map_component.short_name
+        adoptions = Page.get_adopting_facilities(@map_component)
+        build_map_component adoptions
+      end
```

Comment:
small request: could you move this to a private method? You'll see I end up doing a similar process of iterating through PageComponents and then pull it out in this refactor: in https://github.com/department-of-veterans-affairs/diffusion-marketplace/pull/544/commits/2b7f53995504d83924ffc963f831c357694cd81a

Reply:
Yeah no problem.  Done.